pkg/controller/secrets/secrets.go

```diff
@@ -1,29 +1,49 @@
 package secrets
 
 import (
-	"github.com/openshift/windows-machine-config-operator/pkg/controller/signer"
-	wkl "github.com/openshift/windows-machine-config-operator/pkg/controller/wellknownlocations"
+	"context"
+
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/ssh"
 	core "k8s.io/api/core/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/signer"
 )
 
 const (
 	userDataSecret    = "windows-user-data"
 	userDataNamespace = "openshift-machine-api"
+	PrivateKeySecret  = "cloud-private-key"
 )
 
+// GetPrivateKey fetches the specified secret and extracts the private key data
+func GetPrivateKey(secret kubeTypes.NamespacedName, c client.Client) ([]byte, error) {
+	// Fetch the private key secret
+	privateKeySecret := &core.Secret{}
+	if err := c.Get(context.TODO(), secret, privateKeySecret); err != nil {
+		// Error reading the object - requeue the request.
+		return []byte{}, err
+	}
+	privateKey, ok := privateKeySecret.Data["private-key.pem"]
```

Comment:
Sorry I should have caught this before. Please make `private-key.pem` a constant.